### PR TITLE
Core/storybook linting

### PR DESCRIPTION
**Pull request by clairesunstudio**
Add linting to Core and Core CircleCI build task 

To run lint:
- cd into packages/core
- `rushx lint` or `rushx lint-fix` 

Fixed all lint errors and added missing dependencies for core

### CODE REVIEW

**Comment by clairesunstudio** [comment]
@smurrayatwork this is having issues with using the published version of mayflower-assets instead of the local package... 
In the core branch, button-copy.scss exist but not in the published package, how does rush determine which version to use what versions of mayflower-react and mayflower-assets should be specified in package.json?

```
SassError: Can't find stylesheet to import.
   ╷
15 │ @use '@massds/mayflower-assets/scss/01-atoms/button-copy';
```

File: packages/core/stories/tokens/colors/ColorDisplay.js [hunk 1]
```diff
@@ -3,12 +3,19 @@ import PropTypes from 'prop-types';
 import ButtonCopy from '@massds/mayflower-react/dist/ButtonCopy';
 import './_color-display.scss';
```
Comment (smurrayatwork) [inline packages/core/stories/tokens/colors/ColorDisplay.js:4]: Why is this import here?

**Review by smurrayatwork — APPROVED** [review]
